src/ui/pages/Menu/Menu.tsx

```diff
@@ -85,52 +87,66 @@ const Menu = () => {
-      additionalButtons={<AdditionalButtons />}
-    >
-      <IonGrid>
-        <IonRow className="menu-input-row">
-          <MenuItem
-            index={0}
-            icon={personCircleOutline}
-            label={`${i18n.t("menu.tab.items.profile")}`}
-          />
-          <MenuItem
-            index={1}
-            icon={walletOutline}
-            label={`${i18n.t("menu.tab.items.crypto")}`}
-          />
-        </IonRow>
-        <IonRow className="menu-input-row">
-          <MenuItem
-            index={2}
-            icon={peopleOutline}
-            label={`${i18n.t("menu.tab.items.connections")}`}
-          />
-          <MenuItem
-            index={3}
-            icon={chatbubbleOutline}
-            label={`${i18n.t("menu.tab.items.p2p")}`}
-          />
-        </IonRow>
-        <IonRow className="menu-input-row">
-          <MenuItem
-            index={4}
-            icon={fingerPrintOutline}
-            label={`${i18n.t("menu.tab.items.identity")}`}
-          />
-          <MenuItem
-            index={5}
-            icon={idCardOutline}
-            label={`${i18n.t("menu.tab.items.credentials")}`}
-          />
-        </IonRow>
-      </IonGrid>
-    </TabLayout>
+    <>
+      <TabLayout
+        pageId={pageId}
+        header={true}
+        title={`${i18n.t("menu.tab.header")}`}
+        additionalButtons={<AdditionalButtons />}
+      >
+        <IonGrid>
+          <IonRow className="menu-input-row">
+            <MenuItem
+              index={1}
+              icon={personCircleOutline}
+              label={`${i18n.t("menu.tab.items.profile")}`}
+            />
+            <MenuItem
+              index={2}
+              icon={walletOutline}
+              label={`${i18n.t("menu.tab.items.crypto")}`}
+            />
+          </IonRow>
+          <IonRow className="menu-input-row">
+            <MenuItem
+              index={3}
+              icon={peopleOutline}
+              label={`${i18n.t("menu.tab.items.connections")}`}
+            />
+            <MenuItem
+              index={4}
+              icon={chatbubbleOutline}
+              label={`${i18n.t("menu.tab.items.p2p")}`}
+            />
+          </IonRow>
+          <IonRow className="menu-input-row">
+            <MenuItem
+              index={5}
+              icon={fingerPrintOutline}
+              label={`${i18n.t("menu.tab.items.identity")}`}
+            />
+            <MenuItem
+              index={6}
+              icon={idCardOutline}
+              label={`${i18n.t("menu.tab.items.credentials")}`}
+            />
+          </IonRow>
+        </IonGrid>
+      </TabLayout>
+      <SubMenu
+        showSubMenu={showSubMenu}
+        setShowSubMenu={setShowSubMenu}
+        title={`${i18n.t(subMenuTitle[selectedOption])}`}
+        additionalButtons={subMenuAdditionalButtons[selectedOption]}
+      >
+        {SubMenuChildren[selectedOption] && SubMenuChildren[selectedOption]()}
```

Comment:
interesting that you are able to dynamically render the react component as if it were a function with `()` at the end - feature of functional components?

Reply:
First time I use that 😄 

Here is the explanation:

> In the expression `SubMenuChildren[selectedOption]()`, the () is used to invoke or call the function that is stored at the index `selectedOption` in the `SubMenuChildren` array.
>
>Here's an explanation of each part of the expression:
>
>`SubMenuChildren`: This is an array that contains functions, where each function represents a component.
>
>`[selectedOption]`: This part accesses the function at the index specified by `selectedOption` in the `SubMenuChildren` array.
>
>(): The parentheses () are used to call or invoke the function that is retrieved from the array. By adding () after the function reference, you are executing the function and returning the React element that it generates.
>
>In React, components are essentially functions that return React elements (JSX). So, by calling `SubMenuChildren[selectedOption]()`, you are executing the function stored at the `selectedOption` index in the `SubMenuChildren` array and rendering the React element that the function returns.
>
>This approach allows you to dynamically render different components based on the value of `selectedOption` by invoking the corresponding function from the array.

Reply:
Mhm, functional components - makes sense - cool :)